migrations/m9_remove_idea_bot.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ *
+ * Ideas extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\ideas\migrations;
+
+class m9_remove_idea_bot extends \phpbb\db\migration\migration
+{
+	public function effectively_installed()
+	{
+		return !$this->config->offsetExists('ideas_poster_id');
+	}
+
+	static public function depends_on()
+	{
+		return [
+			'\phpbb\ideas\migrations\m1_initial_schema',
+			'\phpbb\ideas\migrations\m3_acp_data',
+			'\phpbb\ideas\migrations\m4_update_statuses',
+			'\phpbb\ideas\migrations\m6_migrate_old_tables',
+			'\phpbb\ideas\migrations\m7_drop_old_tables',
+		];
+	}
+
+	public function update_data()
+	{
+		return [
+			['custom', [[$this, 'update_topic_authors']]],
+			['config.remove', ['ideas_poster_id']],
+		];
+	}
+
+	/**
+	 * Replace the Ideas Bot stored in the posts and topics tables with the
+	 * original author's information. Bot gone, order restored to universe.
+	 */
+	public function update_topic_authors()
+	{
+		// Return if the Ideas Bot does not exist at this point for some reason.
+		if (!$this->config->offsetExists('ideas_poster_id'))
+		{
+			return;
+		}
+
+		// Get real author info for ideas that were posted by the Ideas Bot
+		$topics = [];
+		$sql_array = [
+			'SELECT'	=> 'i.topic_id, i.idea_author, u.username, u.user_colour, t.topic_first_post_id',
+			'FROM'		=> [
+				$this->table_prefix . 'ideas_ideas'	=> 'i',
+			],
+			'LEFT_JOIN'	=> [
+				[
+					'FROM'	=> [$this->table_prefix . 'topics' => 't'],
+					'ON'	=> 't.topic_id = i.topic_id',
+				],
+				[
+					'FROM'	=> [$this->table_prefix . 'users' => 'u'],
+					'ON'	=> 'u.user_id = i.idea_author',
+				],
+			],
+			'WHERE'		=> 't.topic_poster = ' . (int) $this->config->offsetGet('ideas_poster_id'),
+		];
+
+		$sql = $this->db->sql_build_query('SELECT', $sql_array);
+		$result = $this->db->sql_query($sql);
+		while ($row = $this->db->sql_fetchrow($result))
+		{
+			$topics[$row['topic_id']] = [
+				'topic_poster_id'     => $row['idea_author'],
+				'topic_poster_name'   => $row['username'],
+				'topic_poster_colour' => $row['user_colour'],
```

Comment:
The `username` and `user_colour` come back as `NULL` when the user does not exist (deleted) and these target columns don't allow nulls. You'll need to determine if that's the case and handle accordingly, assigning empty strings to `topic_poster_name` and `topic_poster_colour` and you'll also need to set `topic_poster_id` to the `ANONYMOUS` constant value (looks like it's 1 in the constants file).